### ISSUE TITLE
Brief Review: Supervised Contrastive Learning.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#55]
Sik-Ho Tsang. [Brief Review: Supervised Contrastive Learning](https://sh-tsang.medium.com/brief-review-supervised-contrastive-learning-d25c6d2065c2).

**Comment by NorbertZheng** [comment]
## Overview
SupCon, Supervised Contrastive Loss, Outperforms Cross Entropy Loss?

![image](https://user-images.githubusercontent.com/37235023/216515562-0a43222d-3c38-425f-a137-8cc738e85426.png)
SupCon loss consistently outperforms cross-entropy with standard data augmentations.

Supervised Contrastive Learning (SupCon), by Google Research, Boston University, and MIT. 2020 NeurIPS, Over 1200 Citations.

**Comment by NorbertZheng** [comment]
## Supervised Contrastive Learning (SupCon)
![image](https://user-images.githubusercontent.com/37235023/216515800-de0a2b90-f6c4-4cee-a744-e384da9c4439.png)
Cross entropy, self-supervised contrastive loss and supervised contrastive loss.
 - **(a) The cross entropy loss (left)**: uses labels and a softmax loss to train a classifier;
 - **(b) The self-supervised contrastive loss (middle)**: uses a contrastive loss and data augmentations to learn representations, such as SimCLR:
   ![image](https://user-images.githubusercontent.com/37235023/216516168-861c4cf3-54c0-4c98-bfdf-294751ec2f7e.png)
 - (It is assumed that contrastive learning is known already.)
 - **(c) The proposed supervised contrastive loss (right)**: also learns representations using a contrastive loss, but **uses label information to sample positives in addition to augmentations of the same image**.

This proposed loss contrasts the set of all samples from the same class as positives against the negatives from the remainder of the batch, using the labels.
 - Both contrastive methods (b) and (c) can have an optional second stage which trains a model on top of the learned representations.

![image](https://user-images.githubusercontent.com/37235023/216516448-02f781c1-9512-45a2-9eeb-99f36455402f.png)
Supervised vs. self-supervised contrastive losses. As demonstrated by the photo of the black and white puppy, taking class label information into account results in an embedding space where elements of the same class are more closely aligned than in the self-supervised case, even the appearance are not the same. Because they are coming from the same class based on the supervised labels.

There are two variants $L_{out}^{sup}$ and $L_{in}^{sup}$:
![image](https://user-images.githubusercontent.com/37235023/216516906-8e3e153e-4c8b-4a60-a369-43cb5ae23559.png)
 - $L_{out}^{sup}$: The summation over positives is located outside of the log.
 - $L_{in}^{sup}$: The summation is located inside of the log.
 - Since the log function is concave function, Jensen’s Inequality suggests that:

$$
L_{in}^{sup} \leq L_{out}^{sup},
$$

 - we can see that $L_{out}^{sup}$ is the upper bound.
 - And the results also show that $L_{out}^{sup}$ outperforms $L_{in}^{sup}$. Only $L_{out}^{sup}$ is used in the experiment.
   ![image](https://user-images.githubusercontent.com/37235023/216517425-4f845236-565f-4441-8ddf-591661e6276e.png)
   ImageNet Top-1 classification accuracy for supervised contrastive losses on ResNet-50 for a batch size of 6144.

**Comment by NorbertZheng** [comment]
## Results
### Image Classification
![image](https://user-images.githubusercontent.com/37235023/216517632-d0de5a57-5dea-413d-b2fa-17e318edfa53.png)
Top-1 classification accuracy on ResNet-50 for various datasets.

 - SupCon generalizes better than cross-entropy, margin classifiers (with use of labels) and unsupervised contrastive learning techniques on CIFAR-10, CIFAR-100 and ImageNet datasets.

![image](https://user-images.githubusercontent.com/37235023/216517782-34325410-13e0-42d1-a1c8-c195d946b388.png)
Top-1/Top-5 accuracy results on ImageNet for AutoAugment with ResNet-50 and for Stacked RandAugment with ResNet-101 and ResNet-200.

 - A new state of the art accuracy of 78.7% on ResNet-50 with AutoAugment is achieved.

### Robustness
![image](https://user-images.githubusercontent.com/37235023/216517911-c985b4f3-dfb9-48b7-948c-5effd7dc014c.png)
Training with supervised contrastive loss makes models more robust to corruptions in images.

 - The SupCon models have lower mCE values across different corruptions. SupCon models demonstrate lesser degradation in accuracy with increasing corruption severity.

### Ablation Study
![image](https://user-images.githubusercontent.com/37235023/216518020-36bfa7e8-e91f-4f95-abc8-56e3928ee613.png)
Accuracy against (a) Hyperparameters, (b) Batch Size, (c) Epochs, and (d) Temperature.

### Transfer Learning
![image](https://user-images.githubusercontent.com/37235023/216518091-3fb336cf-765e-4006-86fb-88f992f364e7.png)
Numbers are mAP for VOC2007; mean-per-class accuracy for Aircraft, Pets, Caltech, and Flowers; and top-1 accuracy for all other datasets.

 - SupCon is on par with cross-entropy and self-supervised contrastive loss on transfer learning performance when trained on the same architecture.
 - Understanding the connection between training objective and transfer performance is left to future work.

**Comment by NorbertZheng** [comment]
Reference
[2020 NeurIPS] [SupCon] [Supervised Contrastive Learning](https://arxiv.org/abs/2004.11362).

**Comment by NorbertZheng** [comment]
## Image Classification
1989 … 2020 [SupCon] … 2022 [[ConvNeXt](https://sh-tsang.medium.com/review-convnext-a-convnet-for-the-2020s-53b9ada30ab9)] [[PVTv2](https://sh-tsang.medium.com/review-pvtv2-improved-baselines-with-pyramid-vision-transformer-5cfd354a53bd)] [[ViT-G](https://sh-tsang.medium.com/review-scaling-vision-transformers-fc11d867ada6)] [[AS-MLP](https://sh-tsang.medium.com/review-as-mlp-an-axial-shifted-mlp-architecture-for-vision-a59c7fcbb3e6)] [[ResTv2](https://sh-tsang.medium.com/brief-review-rest-v2-simpler-faster-and-stronger-8de1ec5c0)]
